lib/components/app-service-bot.js

```diff
@@ -50,48 +56,60 @@ AppServiceBot.prototype.getJoinedRooms = function() {
  * @return {Promise<Object,Error>} Resolves to a map of user ID => display_name avatar_url
  */
-AppServiceBot.prototype.getJoinedMembers = function(roomId) {
+AppServiceBot.prototype.getJoinedMembers = function(roomId, storeMembership) {
     return this.client._http.authedRequestWithPrefix(
         undefined, "GET", "/rooms/" + encodeURIComponent(roomId) + "/joined_members",
         undefined, undefined, "/_matrix/client/r0"
-    ).then(function(res) {
+    ).then((res) => {
         if (!res.joined) {
             return {};
         }
+
+        if(storeMembership === true) {
+            for(const member in res.joined) {
+                if (this.isRemoteUser(member)) {
+                    this._setJoined(member, roomId);
+                }
+            }
+        }
         return res.joined;
     });
 };
 
 /**
- * Get the membership lists of every room the AS bot is in. This does a /sync.
- * @return {Promise} Resolves to a dict of room_id: {@link AppServiceBot~RoomInfo}.
+ * @throws {Error} This will always throw because /sync is no longer supported.
  */
 AppServiceBot.prototype.getMemberLists = function() {
-    var self = this;
-    var filterJson = JSON.stringify({
-        event_fields: ["content.membership", "type", "state_key"],
-        presence: {
-            not_types:["*"]
-        },
-        room: {
-            timeline: { limit: 0 },
-            ephemeral: { not_types: ["*"] },
-            state: { types: ["m.room.member"] }
-        }
-    });
-    return this.client._http.authedRequestWithPrefix(
-        undefined, "GET", "/sync", {filter: filterJson}, undefined, "/_matrix/client/r0"
-    ).then(function(res) {
-        var rooms = res.rooms || {};
-        var roomIdToRoom = rooms.join || {};
-        var dict = {};
-        Object.keys(roomIdToRoom).forEach(function(roomId) {
-            dict[roomId] = self._getRoomInfo(roomId, roomIdToRoom[roomId]);
-        })
-        return dict;
-    });
+    // var self = this;
+    // var filterJson = JSON.stringify({
+    //     event_fields: ["content.membership", "type", "state_key"],
+    //     presence: {
+    //         not_types:["*"]
+    //     },
+    //     room: {
+    //         timeline: { limit: 0 },
+    //         ephemeral: { not_types: ["*"] },
+    //         state: { types: ["m.room.member"] }
+    //     }
+    // });
+    // return this.client._http.authedRequestWithPrefix(
+    //     undefined, "GET", "/sync", {filter: filterJson}, undefined, "/_matrix/client/r0"
+    // ).then(function(res) {
+    //     var rooms = res.rooms || {};
+    //     var roomIdToRoom = rooms.join || {};
+    //     var dict = {};
+    //     Object.keys(roomIdToRoom).forEach(function(roomId) {
+    //         dict[roomId] = self._getRoomInfo(roomId, roomIdToRoom[roomId]);
+    //     })
+    //     return dict;
+    // });
+    throw new Error(
+        "The ASBot can no longer /sync because this functionality was removed." +
```

Comment:
s/ASBot/appservice bot/ please. Spelling things out is better than acronyms in my books.